src/DropdownButton.jsx

```diff
@@ -111,6 +111,11 @@ var DropdownButton = React.createClass({
     e.preventDefault();
 
     this.setDropdownState(!this.state.open);
+    if(this.props.onClick && this.props.noCaret ){
```

Comment:
I'm not following the `noCaret` check for this either. To me the `noCaret` should only be used as presentation logic to toggle whether the caret is displayed or not. It should not have an effect on clicking. If it is causing a problem with clicking can you supply a test to prove so.